runtime/lua/nvim/util.lua

```diff
@@ -0,0 +1,113 @@
+local util = {
+  quickfix = {},
+}
+
+util.tostring = function(obj)
+  local stringified = ''
+  if type(obj) == 'table' then
+    stringified = stringified .. '{'
+    for k, v in pairs(obj) do
+      stringified = stringified .. util.tostring(k) .. '=' .. util.tostring(v) .. ','
+    end
+    stringified = stringified .. '}'
+  else
+    stringified = tostring(obj)
+  end
+
+  return stringified
+end
+
+util.handle_uri = function(uri)
+  local file_prefix = 'file://'
+  if string.sub(uri, 1, #file_prefix) == file_prefix then
+    return string.sub(uri, #file_prefix + 1, #uri)
+  end
+
+  return uri
+end
+
+-- Determine whether a Lua table can be treated as an array.
+-- Returns:
+--  true    A non-empty array
+--  false   A non-empty table
+--  nil     An empty table
+util.is_array = function(table)
+  if type(table) ~= 'table' then
+    return false
+  end
+
+  local count = 0
+
+  for k, _ in pairs(table) do
+    if type(k) == "number" then
+      count = count + 1
+    else
+      return false
+    end
+  end
+
+  if count > 0 then
+    return true
+  else
+    return nil
+  end
+end
+
+util.get_file_line = function(file_name, line_number)
+  local f = assert(io.open(file_name, 'r'))
```

Comment:
Lua io module is somewhat unreliable, and not necessary since we have `luv` and Nvim itself. Could use `nvim_call_function('readfile')`, or maybe something in `vim.loop`.